src/spyglass/spikesorting/spikesorting_merge.py

```diff
@@ -50,6 +56,82 @@ class CuratedSpikeSorting(SpyglassMixin, dj.Part):  # noqa: F811
         -> CuratedSpikeSorting
         """
 
+    def get_restricted_merge_ids(
+        self,
+        key: dict,
+        sources: list = ["v0", "v1"],
+        v1_artifact: bool = True,
+        as_dict: bool = False,
+    ):
+        """Helper function to get merge ids for a given interpretable key
+
+        Parameters
+        ----------
+        key : dict
+            restriction for any stage of the spikesorting pipeline
+        sources : list, optional
+            list of sources to restrict to
+        v1_artifact : bool, optional
+            whether to restrict to v1 artifact intervals, by default True
+        as_dict : bool, optional
+            whether to return merge_ids as a list of dictionaries, by default False
+
+        Returns
+        -------
+        merge_ids : list
+            list of merge ids from the restricted sources
+        """
+        merge_ids = []
+
+        if "v1" in sources:
+            key_v1 = key.copy()
+            # Recording restriction
+            table = SpikeSortingRecordingSelection() & key_v1
+            if v1_artifact:
+                # Artifact restriction
+                table_artifact = ArtifactDetectionSelection * table & key_v1
+                artifact_restrict = table_artifact.proj(
+                    interval_list_name="artifact_id"
+                ).fetch(as_dict=True)
+                # convert interval_list_name from artifact uuid to string
+                for key_i in artifact_restrict:
+                    key_i["interval_list_name"] = str(
+                        key_i["interval_list_name"]
+                    )
+                if "interval_list_name" in key_v1:
+                    key_v1.pop(
+                        "interval_list_name"
+                    )  # pop the interval list since artifact intervals are now the restriction
+                # Spike sorting restriction
+                table = (
+                    (SpikeSortingSelection() * table.proj())
+                    & artifact_restrict
+                    & key_v1
+                )
+            else:
+                # use the supplied interval to restrict
+                table = (SpikeSortingSelection() * table.proj()) & key_v1
+            # Metric Curation restriction
+            table = (MetricCurationSelection() * table) & key_v1
+            # get curations
+            table = (CurationV1() * table) & key_v1
+            table = SpikeSortingOutput().CurationV1() & table
+            merge_ids.extend(table.fetch("merge_id", as_dict=as_dict))
+
+        if "v0" in sources:
+            key_v0 = key.copy()
+            if "sort_interval" not in key_v0:
+                if "interval_list_name" in key_v0:
+                    key_v0["sort_interval"] = key_v0["interval_list_name"]
+                    key_v0.pop("interval_list_name")
```

Comment:
I prefer anonymous assignment (`_ = thing`) when using `pop` to signal intent to ditch it
```suggestion
            if "sort_interval" not in key_v0 and "interval_list_name" in key_v0:
                key_v0["sort_interval"] = key_v0["interval_list_name"]
                _ = key_v0.pop("interval_list_name")
```